### PR TITLE
 Dp469 : Project-wise Nominations Dashboard for Sourcing Org

### PR DESCRIPTION
**Pull request by Nilesh5995**

### CODE REVIEW

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html [hunk 1]
```diff
@@ -36,7 +36,7 @@ <h5 class="m-0 p-0 font-weight-bold d-inline-block">{{ programDetails?.name }} <
                     <label class="mb-0">{{resourceService?.frmelmnts?.lbl?.contentType}}</label>
                     <p class="sb__ellipsis sb__ellipsis--one cursor-pointer" suiPopup popupHeader="{{resourceService.frmelmnts.lbl.contentType}}" popupText="{{ programContentTypes }}"><span>{{ programContentTypes }}</span></p>
 			</div>
-            <div *ngIf="!isContributionDashboardTabActive" class="sb-meta-data-container sb-meta-data-column px-10">
+            <!-- <div *ngIf="!isContributionDashboardTabActive" class="sb-meta-data-container sb-meta-data-column px-10">
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html:39]: @Nilesh5995  Do not comment out the code. Remove if not required

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html [hunk 2]
```diff
@@ -271,13 +305,12 @@ <h5 class="m-0 p-0 font-weight-bold d-inline-block">{{ programDetails?.name }} <
...
-            <div class="ui segment p-0 m-0 mt-4 sb-bg-color-white dynamicHeight" suiTabContent="3">
+            <!-- <div class="ui segment p-0 m-0 mt-4 sb-bg-color-white dynamicHeight" suiTabContent="3">
 				<div class="ui container">
 					<div class="pt-24">
 					  <div class="d-flex flex-ai-center flex-jc-space-between sb-pageSection">
-						<div class="sb-pageSection-header">
-						  <h4 class="sb-pageSection-title"> {{userService.userProfile?.organisations[0]?.organisationId}}</h4>
-						  <span class="sb-pagSection-count sb-label sb-label-x sb-label-warning">{{sourcingOrgUser.length || 0}}</span>
+						<div class="sb-pageSection-header mb-15">
+						  <h4 class="sb-pageSection-title"> {{userService.userProfile?.organisations[0]?.orgName}}</h4>
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html:313]: @Nilesh5995  where are you showing the no of users of the orgnisation?

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html [hunk 3]
```diff
@@ -309,8 +342,7 @@ <h4 class="sb-pageSection-title"> {{userService.userProfile?.organisations[0]?.o
 							<td>{{orgUser?.framework?.gradeLevel?.join(', ')}}</td>
 							<td>{{orgUser?.framework?.subject?.join(', ')}}</td>
 							<td class="text-center">
-							        <sui-select class="selection pt-5" [(ngModel)]="orgUser.selectedRole" labelField="name" [options]="options" valueField="name"
-									(selectedOptionChange)="onRoleChange()">
+							        <sui-select class="selection pt-5" [(ngModel)]="orgUser.selectedRole" labelField="name" [options]="options" valueField="name" (selectedOptionChange)="onRoleChange()" [isDisabled]="!canAssignUsers">
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html:345]: why these changes are in your PR?

Comment (twsvaishali) [reply src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html:345]: @Nilesh5995

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 4]
```diff
@@ -204,16 +215,42 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
...
+              }
+           });
+           _.forEach(res.content_types, (content_types) => {
+            if (!_.includes(this.nominatedcontentTypes, content_types)) {
+              this.nominatedcontentTypes.push(content_types);
+            }
+         });
         });
       }
+      this.counts.totalNomination = this.counts.pendingNomination +  this.counts.approvedNomination + this.counts.rejectedNomination;
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:251]: this.counts.totalNomination should be  this.nominationsCount. You do not have to use another variable

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 5]
```diff
@@ -339,10 +386,14 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
...
 
   getProgramDetails() {
+    this.counts.totalTextbook = 0;
+    this.counts.totalContentType = 0;
     this.fetchProgramDetails().subscribe((programDetails) => {
       this.programDetails = _.get(programDetails, 'result');
       this.programContext = this.programDetails;
       this.collectionsCount = _.get(this.programDetails, 'collection_ids').length;
+      this.counts.totalTextbook = this.collectionsCount ;
+      this.counts.totalContentType = _.get(this.programDetails, 'content_types').length;
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:396]: this.counts.totalContentType is this.programContentTypes.length. No need to use other var

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 6]
```diff
@@ -339,10 +386,14 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
   }
 
   getProgramDetails() {
+    this.counts.totalTextbook = 0;
+    this.counts.totalContentType = 0;
     this.fetchProgramDetails().subscribe((programDetails) => {
       this.programDetails = _.get(programDetails, 'result');
       this.programContext = this.programDetails;
       this.collectionsCount = _.get(this.programDetails, 'collection_ids').length;
+      this.counts.totalTextbook = this.collectionsCount ;
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:395]: use this.collectionsCount ; only. Why to add another var?

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 7]
```diff
@@ -388,47 +439,6 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
     }));
   }
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:441]: There are already var  this.pendingCount  this.approvedCount this.rejectedCount.. Use these only

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 8]
```diff
@@ -204,16 +212,44 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
...
+          if (res.status === 'Pending') {
+            this.pendingCount = this.pendingCount + 1;
+          }
+          if (res.status === 'Approved') {
+            this.approvedCount  = this.approvedCount + 1;
+          }
+          if (res.status === 'Rejected') {
+            this.rejectedCount = this.rejectedCount + 1;
+          }
+          console.log(res, 'this is res');
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:236]: remove console logs

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 9]
```diff
@@ -204,16 +212,44 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
...
+          _.forEach(res.collection_ids, (collectionId) => {
+              if (!_.includes(textbooks, collectionId)) {
+                textbooks.push(collectionId);
+              }
+           });
+           _.forEach(res.content_types, (content_types) => {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:242]: can not you merge two arrays? instead of doing foreach?

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 10]
```diff
@@ -204,16 +212,44 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
...
           }
-
           if (name) {
             this.nominations.push({
               'name': name.trim(),
               'type': isOrg ? 'Organisation' : 'Individual',
               'nominationData': res
             });
           }
+          if (isOrg) {
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:222]: try using arrow functions for this

Comment (twsvaishali) [reply src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:222]: (org) ?  this.contributedByOrganisation++ : this.contributedByIndividual++ try using this.. @Nilesh5995

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html [hunk 11]
```diff
@@ -36,27 +36,6 @@ <h5 class="m-0 p-0 font-weight-bold d-inline-block">{{ programDetails?.name }} <
...
-            <div *ngIf="!isContributionDashboardTabActive" class="sb-meta-data-container sb-meta-data-column px-10">
-                <label class="width-100">{{ resourceService?.frmelmnts?.lbl?.nominations }} </label>
-                <div class="d-flex">
-                    <dl class="sb-meta-data">
-                        <dt>{{ pendingCount }}</dt>
-                        <dd>{{ resourceService?.frmelmnts?.lbl?.pending }}</dd>
-                    </dl>
-                    <dl class="sb-meta-data">
-                        <dt>{{ approvedCount }}</dt>
-                        <dd>{{ resourceService?.frmelmnts?.lbl?.approved }}</dd>
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.html:48]: This code is needed. Please revert it.

File: src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts [hunk 12]
```diff
@@ -388,47 +423,6 @@ export class ProgramNominationsComponent implements OnInit, AfterViewInit, OnDes
...
-        request: {
-          filters: {
-            program_id: this.programId,
-            status: ['Pending', 'Approved', 'Rejected']
-          },
-          facets: ['program_id', 'status']
-        }
-      }
-    };
-    return this.programsService.post(req);
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:429]: @twsvaishali we don't need these functions?

Comment (twsvaishali) [reply src/app/client/src/app/modules/program/components/program-nominations/program-nominations.component.ts:429]: @venkateshwarans  Not now because, we are having the counts by the list API only. So we can remove this

File: src/app/resourcebundles/data/consumption/en.properties [hunk 13]
```diff
@@ -1351,6 +1351,12 @@ frmelmnts.lbl.contentName = Name
 frmelmnts.lbl.contentNameRequired = Name required
 frmelmnts.lbl.contentNameLength = * Characters in name should be less than
 frmelmnts.lbl.characterLeft = Characters left
+#Nomination Dashboard
+frmelmnts.lbl.contributedBy=Contributed by
+frmelmnts.lbl.nominated=Nominated
+frmelmnts.lbl.uploaded=Uploaded
+frmelmnts.lbl.samples=Samples
+frmelmnts.lbl.contentType=Content Type
```
Comment (twsvaishali) [inline src/app/resourcebundles/data/consumption/en.properties:1359]: this frmelmnts.lbl.contentType string is already in the file. Remove from here
